zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/util/XmlUtil.java

```diff
@@ -51,16 +71,66 @@ public String getFlowNodeName(final FlowNodeInstanceEntity flowNode) {
         flowNode.flowNodeId());
   }
 
+  public HashMap<Long, Map<Long, String>> getUserTasksNames(final List<UserTaskEntity> userTasks) {
+    return getNamesForEntities(
+        userTasks,
+        UserTaskEntity::processDefinitionKey,
+        UserTaskEntity::userTaskKey,
+        this::getUserTaskName);
+  }
+
   public String getUserTaskName(final UserTaskEntity userTask) {
     final var processDefinition =
         processDefinitionServices.getByKey(userTask.processDefinitionKey());
+    return getUserTaskName(processDefinition, userTask);
+  }
+
+  protected String getUserTaskName(
+      final ProcessDefinitionEntity processDefinition, final UserTaskEntity userTask) {
     return extractFlowNodeName(
         processDefinition.bpmnXml(),
         processDefinition.processDefinitionId(),
         "userTask",
         userTask.elementId());
   }
 
+  protected <T> HashMap<Long, Map<Long, String>> getNamesForEntities(
```

Comment:
🔧 `private <T> Map<Long, Map<Long, String>>`

Reply:
I think this one is used in tests. I could change it to package private.

Reply:
It is better to only unit test public methods, and to not change the visibility of private methods for that. If you still need to do so, keep it with default modifier (package visibility) and annotate it with @VisibleForTesting

Reply:
Changed it to a public method.

Reply:
you mean it is changed to private

Reply:
Yes, the method is now private and the test uses a public method.